Makefile

```diff
@@ -1,17 +1,29 @@
-dev-build:
+help:
+		@echo "Jupiter Broadcasting Hugo Project"
+		@echo 
+		@echo "Help for using this Makefile"
+		@echo
+		@echo "For detailed help about this project please visit:"
+		@echo "https://github.com/JupiterBroadcasting/jupiterbroadcasting.com"
+		@echo
+		@echo "------------------------------------------------------------------"
+		@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort  | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-20s\033[0m - %s\n", $$1, $$2}'
```

Comment:
Reviewing #478 and just wanted to identify what you're doing on this, personally and for others. (going to explain what I understand so far and @timlinux could you explain what I don't or I get wrong? :upside_down_face: 

So the grep command (`grep -E`) is using [extended regex](https://www.computernetworkingnotes.com/linux-tutorials/use-extended-regular-expressions-with-grep-command.html), which is more portable than PCRE since it's POSIX compliant (IIRC).

regex explained:
* This regex `^[a-zA-Z_-]+` is looking for lines that start with any alpha character + `-` & `_` and the `+` says one or more consecutive characters. 
* then there's the literal character of `:`
* after that it's a little loose with `.*?` and can actually be replaced with something like: `[a-zA-Z_ -]+`, but either way it works
* this matches up till the `##` which is the comments right after each make target, which this means all make targets in the future will require a comment to be included in the help section
* ` .*$$` - this then match a literal space character and anything after that (the comment) till the end, which requires 2 `$`, so make doesn't interpret the `$` as something else.

The `$(MAKEFILE_LIST)` appears to list the possible make files (which if we had more should grab all files), but for now it just lists `Makefile`

The sort just sorts the make targets (which sort should just do alphanumeric sort), but could be forced with `-d` which is a dictionary sort (alphanumeric).

Then awk is being used to print the output from the grep command.
* `BEGIN` - I'm not super familiar with `awk`, so I don't know what this does
* Next is the field separator (`FS`) begin defined (or that's what google says :upside_down_face:), which means all the content between the actual commands and the `##` are used to essentially split the content as a delimiter
* the printf command is how the content is getting printed out
* the `\033`, `[36m`, and `[0m` all appear to be color escape things (don't exactly know how it works, but I've seen it in scripts before
* the `%-20s` is essentially saying print a string (typically `%s`) and it's "width" is 20 characters, but fill it with spaces if there's not that many characters
* then it's the literal characters of ` - `
* follows by the next string: `%s`
* followed by a newline character `\n`
* the `, $$1, $$2` says it should print the first string and the second string respectively to their positions, and the `$$` is used for ensuring make doesn't interpret those `$` characters.